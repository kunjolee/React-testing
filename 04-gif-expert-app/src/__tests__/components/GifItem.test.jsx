import { render, screen } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Tests on <GifItem/>', () => {
    const url = 'https://sample.com/onepunch.gif';
    const title = 'onepunch';
    it('should match the snapshot', () => {
        const { container } = render(<GifItem title='My title' url='My url' />);
        expect(container).toMatchSnapshot();
    });

    it('should show the image with the correct url and alt', () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        // screen.getByRole('img').src
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    it('should show the title in the component', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});
