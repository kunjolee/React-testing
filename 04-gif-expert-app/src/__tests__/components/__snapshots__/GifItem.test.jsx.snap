// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem/> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="My title"
      src="My url"
    />
    <p>
      My title
    </p>
  </div>
</div>
`;
